packages/openapi-generator/src/cli/index.ts

```diff
@@ -25,98 +28,19 @@ $ openapi-generator --input ./my-spec.yaml --outputDir ./client --transpile`
-      description:
-        'Remove all files in the output directory before generation. Be cautious when using this option, as it really removes EVERYTHING in the output directory.',
-      default: false
-    }),
-    skipValidation: flags.boolean({
-      description:
-        'By default, the generation fails, when there are duplicate or invalid names for operations and/or path parameters after transforming them to camel case. Set this to true to enable unique and valid name generation. The names will then be generated by appending numbers and prepending prefixes.',
-      default: false
-    }),
-    tsConfig: flags.string({
-      description:
-        'Replace the default `tsconfig.json` by passing a path to a custom config. By default, a `tsconfig.json` is only generated, when transpilation is enabled (`--transpile`). If a directory is passed, a `tsconfig.json` file is read from this directory.',
-      parse: input => resolve(input),
-      helpValue: '<path/to/tsconfig.json>'
-    }),
-    packageJson: flags.boolean({
-      description:
-        'When enabled, a `package.json`, that specifies dependencies and scripts for transpilation and documentation generation is generated.',
-      default: false
-    }),
-    verbose: flags.boolean({
-      description: 'Turn on verbose logging.',
-      default: false
-    }),
-    optionsPerService: flags.string({
-      description:
-        'Set the path to a file containing the options per service. The configuration allows to set a `directoryName` and `packageName` for every service, identified by the path to the original file. It also makes sure that names do not change between generator runs. If a directory is passed, a `options-per-service.json` file is read/created in this directory.',
-      helpValue: '<path/to/options-per-service.json>',
-      parse: input => {
-        const isFilePath =
-          (existsSync(input) && lstatSync(input).isFile()) || !!extname(input);
-        return isFilePath
-          ? resolve(input)
-          : resolve(input, 'options-per-service.json');
-      }
-    }),
-    packageVersion: flags.string({
-      description: 'Set the version in the generated package.json.',
-      default: '1.0.0',
-      hidden: true
-    }),
-    readme: flags.boolean({
-      description:
-        'Generate default `README.md` files in the client directories.',
-      default: false,
-      hidden: true
-    }),
-    metadata: flags.boolean({
-      description: 'When enabled, metadata for the API hub is generated.',
-      default: false,
-      hidden: true
-    })
-  };
+  static flags = generatorFlags;
 
   async run(): Promise<void> {
     try {
       const parsed = this.parse(OpenApiGenerator);
-      await generateWithParsedOptions(parsed.flags);
+      if (parsed.flags.config) {
+        await generate({
+          ...(await parseOptionsFromConfig(parsed.flags.config)),
+          ...removeDefaultValues(parsed.flags, parsed.raw)
```

Comment:
I believe it's still necessary, the issue with the raw values is that all paths values (e.g. input, outputDir, ...) aren't resolved (`resolve` function from `path`), if we would use that raw values, we would need to execute `resolve` on all path values again, which would make the `parseFromConfig` function a lot bigger I believe (and also more code duplication since we would resolve twice but only use it once)

Reply:
Got it. Then I have some additional comments:
1. Consider removing the notion of "default values" in the name of this function, consider something like `getSpecifiedFlags` (or maybe you come up with something more fitting).
2. I think you can express `Object.entries(parseRawInput(rawInput))` + the following usage as this: `parsed.raw.map(({flag}) => flag)`. If you pass the result of this to the (currently) `removeDefaultValues` function, you can remove the import of `ParsingTokens` and instead pass a list of keys (`keys: string[]`).

Reply:
Yes just passing a string array is a good idea, but instead of importing the ParsingToken I now need to import FlagToken, because ParsingTokens can theoretically also be ArgTokens and ArgTokens don't have the property "flag" which would lead to a type error if I don't typecast it (AFAIK we don't have ArgTokens so typecasting should be fine)